### ISSUE TITLE
Python.h: No such file or directory

### ISSUE THREAD

**Issue by espressif2022** [kodonnell/qoi#28]
Hello，my CI compilation reports the following error: it cannot find the `python.h` header file. 
It's the same when compiling on Windows. My local Ubuntu `Env` is ok. 
I have a multi-platform compilation `Env` here. When compiling, I will call `qoi.py` to process some images

How can I fix it here?

![企业微信截图_17263054952986](https://github.com/user-attachments/assets/891f2c29-47d8-40ec-83c1-10b9e05ea78f)

**Comment by kodonnell** [comment]
Hmmm, that's interesting - the automated windows builds seem to be fine as per [here](https://github.com/kodonnell/qoi/actions/runs/9632671218/job/26566164250).

It looks like you're compiling for micropython or similar given it's an esp32, is that correct? I've never developed for that - but I assume the missing Python.h is part of that. Normally you'd [just install `python3-dev`](https://stackoverflow.com/a/21530768) to get the headers etc.

Have a look, but it seems like this might be unrelated to this project, and more a general python/micropython question.

**Comment by espressif2022** [comment]
> Hmmm, that's interesting - the automated windows builds seem to be fine as per [here](https://github.com/kodonnell/qoi/actions/runs/9632671218/job/26566164250).
> 
> It looks like you're compiling for micropython or similar given it's an esp32, is that correct? I've never developed for that - but I assume the missing Python.h is part of that. Normally you'd [just install `python3-dev`](https://stackoverflow.com/a/21530768) to get the headers etc.
> 
> Have a look, but it seems like this might be unrelated to this project, and more a general python/micropython question.

Yes, it's esp32, and I used `qoi ` when building the system now. 
I installed python3-dev in my local environment, and it works. 
But I can't seem to install it on our CI compilation server, and I'll try it on Windows or mac again.

Thanks.

**Comment by kodonnell** [comment]
Closing now as I assume this is resolved? Feel free to re-open @espressif2022 if needed.
